shinysdr/test/plugins/wspr/test_demodulator.py

```diff
@@ -162,10 +175,28 @@ def test_frequency_change(self):
         happened on another.
         """
         self.listener.fileOpened('some file')
+        self.assertEqual(self.listener.get_status(), STATUS_RECEIVING)
         self.context.get_absolute_frequency_cell().set(654321)
         self.clockAndSpawn.advance(1)  # Allow cell subscription to fire.
+        self.assertEqual(self.listener.get_status(), STATUS_IDLE)
         self.listener.fileClosed('some file')
         self.assertFalse(self.spawned)
+        self.assertEqual(self.listener.get_status(), STATUS_IDLE)
+
+    def test_decode_runs_long_status(self):
+        """Confirm expected behavior if decode and receive overlap."""
+        self.listener.fileOpened('some file')
+        self.assertEqual(self.listener.get_status(), STATUS_RECEIVING)
+
+        self.listener.fileClosed('some file')
+        self.assertEqual(self.listener.get_status(), STATUS_DECODING)
+
+        self.listener.fileOpened('some file')
+        self.assertEqual(self.listener.get_status(), STATUS_DECODING_AND_RECEIVING)
```

Comment:
The intent was just to deduplicate the contents, but I agree that this is weird and confusing. There now actually is a combined message.